src/main.py

```diff
@@ -25,15 +28,42 @@ def main():
 
         for spotify_track in spotify_playlist.tracks:
             artists = ' '.join(map(lambda artist: artist.name, spotify_track.artists))
-            track_name = f"{spotify_track.title} {artists}"
+            track_name = f'{spotify_track.title} {artists}'
 
+            attemp_count = 1
+            while attemp_count <= 3:
+                try:
+                    yandex_track = yam.search_track(track_name)
+                    yandex_playlist = yam.add_to_playlist(yandex_playlist, yandex_track)
+                    break
+                except TypeError:
+                    print(f'Can\'t found track: "{track_name}" to add to playlist "{spotify_playlist.name}"')
+                    break
+                except Exception as e:
+                    print(f'Something went wrong... Attemp {attemp_count} {e}')
+                
+                attemp_count += 1
+    
+    spotify_playlist = spotify.get_saved_tracks()
+    print(f'Adding {spotify_playlist.name}')
+    
+    for spotify_track in spotify_playlist.tracks:
```

Comment:
Code duplication. 
You can make a function that will take the  `spotify.Playlist` and function which returns new playlist, i.e.: 

```python
def add_playlist(spotify_playlist, yandex_playlist, add_track):
   for spotify_track in spotify_playlist.tracks:
            artists = ' '.join(map(lambda artist: artist.name, spotify_track.artists))
            track_name = f'{spotify_track.title} {artists}'

            for attempt in range(ATTEMPTS_NUMBER):
                try:
                    yandex_track = yam.search_track(track_name)
                    yandex_playlist = add_track(yandex_playlist, yandex_track)
                    break
                except TypeError:
                    print(f'Can\'t found track: "{track_name}" to add to playlist "{spotify_playlist.name}"')
                    break
                except Exception as e:
                    print(f'Something went wrong... Attemp {attempt} {e}')
``` 

So for playlists will be: 

```python
add_playlist(spotify_playlist, yandex_playlist, lambda yandex_playlist, track: yam.add_to_playlist(yandex_playlist, track)
```

For favorites:
```python
add_playlist(spotify_playlist, yandex_playlist, lambda yandex_playlist, track: yam.like_track(track)
```

Reply:
My knowledge of Python is very poor, so you can do better!